### PR TITLE
CEDS 1433 remove and refresh

### PR DESCRIPTION
**Pull request by SteveSugden**
Changes to fix the removal of list items, following by refreshing the page.  This re-submits the remove request and should not fail if the item to be removed is no longer there.

Also stopped referencing list items by index as refreshing the page would cause another item to be removed (or an index out of bounds).

### CODE REVIEW

File: app/controllers/declaration/DestinationCountriesController.scala [hunk 1]
```diff
@@ -174,25 +174,15 @@ class DestinationCountriesController @Inject()(
...
+
+    val updatedCountries = cachedData.countriesOfRouting.filterNot(keys.contains(_))
 
     val updatedCache = cachedData.copy(countriesOfRouting = updatedCountries)
 
     updateCache(journeySessionId, updatedCache)
       .map(_ => Ok(destinationCountriesStandardPage(userInput.discardingErrors, updatedCache.countriesOfRouting)))
   }
 
-  private def parseRemoval(keys: Seq[String]): Try[Int] =
```
Comment (ed0906) [inline app/controllers/declaration/DestinationCountriesController.scala]: 👍

File: app/controllers/declaration/ItemTypeController.scala [hunk 2]
```diff
@@ -193,10 +191,10 @@ class ItemTypeController @Inject()(
 
     val itemTypeUpdated = label.name match {
       case `taricAdditionalCodesKey` =>
-        itemTypeCached.copy(taricAdditionalCodes = removeElement(itemTypeCached.taricAdditionalCodes, label.index))
+        itemTypeCached.copy(taricAdditionalCodes = removeElement(itemTypeCached.taricAdditionalCodes, label.value))
       case `nationalAdditionalCodesKey` =>
         itemTypeCached.copy(
-          nationalAdditionalCodes = removeElement(itemTypeCached.nationalAdditionalCodes, label.index)
+          nationalAdditionalCodes = removeElement(itemTypeCached.nationalAdditionalCodes, label.value)
```
Comment (ed0906) [inline app/controllers/declaration/ItemTypeController.scala]: 👍

File: app/controllers/util/MultipleItemsHelper.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ object MultipleItemsHelper {
     */
   def remove[A](idOpt: Option[String], cachedData: Seq[A]): Seq[A] = idOpt match {
     case Some(id) if validateId(id) && cachedData.length - 1 >= id.toInt => removeItem(id, cachedData)
-    case _                                                               => throw new InternalServerException("Incorrect id")
```
Comment (ed0906) [inline app/controllers/util/MultipleItemsHelper.scala]: 🥇  I noticed we've stopped using this helper in a lot of the controllers, does it not work how we want it to?

File: app/views/components/fields/field_multiple_values.scala.html [hunk 4]
```diff
@@ -42,19 +43,12 @@
     </label>
 
     @if(elements.nonEmpty) {
-      @if(field.name.endsWith("[]")) {
```
Comment (ed0906) [inline app/views/components/fields/field_multiple_values.scala.html]: 🥇

File: app/views/components/fields/field_text_multiple_values.scala.html [hunk 5]
```diff
@@ -24,7 +24,8 @@
     elements: Seq[String] = Seq.empty,
     inputClass: Option[String] = None,
     labelClass: Option[String] = None,
-    includeAddButton: Boolean = true
+    includeAddButton: Boolean = true,
+    valuePrefix: Option[String] = None
```
Comment (ed0906) [inline app/views/components/fields/field_text_multiple_values.scala.html]: What's `valuePrefix` for?

File: app/controllers/util/MultipleItemsHelper.scala (hunk 3 above)
Comment (SteveSugden) [reply app/controllers/util/MultipleItemsHelper.scala]: I've replaced this helper method.  We don't want to be removing by index.

File: app/views/components/fields/field_text_multiple_values.scala.html (hunk 5 above)
Comment (SteveSugden) [reply app/views/components/fields/field_text_multiple_values.scala.html]: Differentiating the ids of different types of list items on the same form.  (Only used in Item Type page at present)

**Comment by SteveSugden** [comment]
test this please

**Pull request merged**